File: scrapper/theverge.py

from bs4 import BeautifulSoup
import requests

import datetime

URL = 'https://www.theverge.com/scooters'

def get_article_html():
    page_html = requests.get( URL )
    soup = BeautifulSoup(page_html.text, 'html.parser')
    page_data = soup.find_all( 'div', { 'class': 'c-entry-box--compact' } )
    return page_data

def extract_data(page_data):
    articles = []
    for article in page_data:
        title = article.find('a', { 'data-chorus-optimize-field': 'hed' })
        url = title['href']
        title = title.string.strip()

        time = article.find('time', { 'class': 'c-byline__item' })
        time = str(time.string.strip())

        image = article.find('div', {'class': 'c-entry-box--compact__image'}).find_all('img')
        image_url = str(image[1]['src']).strip()

        article_obj = { 
            'title': title,
            'published': time,
            'image': image_url,
            'url': url,
            'scrappedAt': datetime.datetime.now(),
        }
        articles.append(article_obj)
    return articles

def get_article_data():
    article_html_list = get_article_html()
    output = extract_data(article_html_list)
    return output

File: frontend/src/components/TheVerge.js
import React from "react";
import "./Article.css";

const TheVerge = ({ title, url, published, image_url }) => {
  return (
    <div className="article">
      <a href={url}>
        <img src={image_url} alt={title} title={title}></img>
      </a>
      <div className="article__data">
        <a className="article__title" href={url}>
          {title}
        </a>
        <h5 className="article__published">{published}</h5>
      </div>
    </div>
  );
};

export default TheVerge;
